File: android/expoview/src/test/java/host/exp/exponent/ABIVersionTest.java
package host.exp.exponent;

import org.junit.Test;

public class ABIVersionTest {
  @Test
  public void unversionedIsBiggerThanSdk32() {
    assert (ABIVersion.toNumber("UNVERSIONED") > ABIVersion.toNumber("32.0.0"));
  }

  @Test
  public void majorUpdateIsBiggerThanNone() {
    assert (ABIVersion.toNumber("32.0.0") > ABIVersion.toNumber("31.0.0"));
  }

  @Test
  public void minorUpdateIsBiggerThanNone() {
    assert (ABIVersion.toNumber("32.1.0") > ABIVersion.toNumber("32.0.0"));
  }

  @Test
  public void patchUpdateIsBiggerThanNone() {
    assert (ABIVersion.toNumber("32.0.1") > ABIVersion.toNumber("32.0.0"));
  }

  @Test
  public void sameVersionsAreEqual() {
    assert (ABIVersion.toNumber("32.0.0") == ABIVersion.toNumber("32.0.0"));
  }

  @Test
  public void minorUpdateIsBiggerThanPatchUpdate() {
    assert (ABIVersion.toNumber("32.1.0") > ABIVersion.toNumber("32.0.1"));
  }
}

File: android/expoview/src/test/java/host/exp/exponent/ABIVersionTest.kt
package host.exp.exponent

import host.exp.exponent.ABIVersion.toNumber
import org.junit.Test

class ABIVersionTest {
  @Test
  fun unversionedIsBiggerThanSdk32() {
    assert(toNumber("UNVERSIONED") > toNumber("32.0.0"))
  }

  @Test
  fun majorUpdateIsBiggerThanNone() {
    assert(toNumber("32.0.0") > toNumber("31.0.0"))
  }

  @Test
  fun minorUpdateIsBiggerThanNone() {
    assert(toNumber("32.1.0") > toNumber("32.0.0"))
  }

  @Test
  fun patchUpdateIsBiggerThanNone() {
    assert(toNumber("32.0.1") > toNumber("32.0.0"))
  }

  @Test
  fun sameVersionsAreEqual() {
    assert(toNumber("32.0.0") == toNumber("32.0.0"))
  }

  @Test
  fun minorUpdateIsBiggerThanPatchUpdate() {
    assert(toNumber("32.1.0") > toNumber("32.0.1"))
  }
}
